### PR TITLE
PI-1157: Implement public key logging

### PR DESCRIPTION
**Pull request by miles-williams-cko**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [ ] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: Source/CheckoutAPIClient.swift [hunk 1]
```diff
@@ -90,7 +90,9 @@ public class CheckoutAPIClient {
         
         let dateProvider = DateProvider()
         let framesEventLogger = FramesEventLogger(checkoutEventLogger: checkoutEventLogger, dateProvider: dateProvider)
-        let networkFlowLoggerFactory = NetworkFlowLoggerFactory(framesEventLogger: framesEventLogger)
+        let networkFlowLoggerFactory = NetworkFlowLoggerFactory(
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:93]: Is this right formatting ?
```
let networkFlowLoggerFactory = NetworkFlowLoggerFactory(framesEventLogger: framesEventLogger,
                                                                                                        publicKey: publicKey)
```

Comment (harry-brown-cko) [reply Source/CheckoutAPIClient.swift:93]: This is different from what we do on 3DS, but I would prefer this style anyway. Can we adopt something similar going forwards?

File: Tests/Core/Logging/NetworkFlowLoggerTests.swift [hunk 2]
```diff
@@ -56,11 +58,12 @@ final class NetworkFlowLoggerTests: XCTestCase {
         
         let subject = NetworkFlowLogger(
             correlationID: "",
+            publicKey: "",
```
Comment (harry-brown-cko) [inline Tests/Core/Logging/NetworkFlowLoggerTests.swift:61]: is it worth testing with values in these fields? Just to check that you're not putting the correlationID into the publicKey field on the logs and visa versa

**Review by harry-brown-cko — COMMENTED** [review]
one comment on tests, otherwise looks good

File: Tests/Core/Logging/NetworkFlowLoggerTests.swift (hunk 2 above)
Comment (miles-williams-cko) [reply Tests/Core/Logging/NetworkFlowLoggerTests.swift]: Whoops, I missed this. Yes, definitely.

**Review by harry-brown-cko — APPROVED** [review]

**Review by daven-gomes-cko — APPROVED** [review]

**Review by ioan-ghisoi-cko — APPROVED** [review]

**Review by deepesh-vasthimal-cko — APPROVED** [review]

**Pull request merged**